lib/src/main/kotlin/org/walletconnect/impls/MoshiPayloadEncryption.kt

```diff
@@ -56,20 +58,34 @@ class MoshiPayloadEncryption(moshi: Moshi) : Session.PayloadEncryption {
     override fun decrypt(encryptedPayloadJson: String, key: String): String {
         val encryptedPayload = encryptedPayloadAdapter.fromJson(encryptedPayloadJson) ?: throw IllegalArgumentException("Invalid json payload!")
 
-        // TODO verify hmac
+        val hexKey = decode(key)
+        val iv = decode(encryptedPayload.iv)
+        val encryptedData = decode(encryptedPayload.data)
+        val providedHmac = decode(encryptedPayload.hmac)
+
+        // verify hmac
+        with(HMac(SHA256Digest())) {
```

Comment:
Would probably be nice of us to contribute this back to the main lib at some point. Maybe just create a cleaned up version to PR off of that library.